indico/modules/rb/client/js/modules/admin/MapAreasPage.jsx

```diff
@@ -0,0 +1,358 @@
+                       onClick={onAreaClick}
+                       id={id}>
+                <Tooltip offset={[bottomRight.x - center.x, bottomRight.y - center.y]}
+                         direction="bottom"
+                         sticky
+                         permanent
+                         interactive>
+                    <>
+                        {name}
+                        {isDefault && (
+                            <span className="area-default">★</span>
+                        )}
+                    </>
+                </Tooltip>
+            </Rectangle>
+        );
+    };
+
+    onSubmit = async (data) => {
+        const {actions: {createMapArea, updateMapAreas}} = this.props;
+        const {areaProps: {id}} = this.state;
+
+        if (id === undefined) {
+            await createMapArea(data);
+        } else {
+            await updateMapAreas([{...data, id}]);
+        }
+
+        this.setState({areaProps: null});
+    };
+
+    renderModalForm = () => {
+        const {areaProps} = this.state;
+        if (!areaProps) {
+            return null;
+        }
+
+        const {id, bounds, name, isDefault} = areaProps;
+        return (
+            <FinalForm onSubmit={this.onSubmit}
+                       render={this.renderForm}
+                       subscription={{}}
+                       initialValues={{
+                           name: id === undefined ? null : name,
+                           bounds,
+                           default: isDefault
+                       }} />
+        );
+    };
+
+    renderForm = ({handleSubmit}) => {
+        const {areaProps: {id}} = this.state;
+        const isNewArea = id === undefined;
+
+        return (
+            <Modal onClose={() => this.setState({areaProps: null})}
+                   closeIcon
+                   open>
+                <Modal.Header>
+                    {isNewArea ? (
+                        <Translate>Create new area</Translate>
+                    ) : (
+                        <Translate>Update area</Translate>
+                    )}
+                </Modal.Header>
+                <Modal.Content>
+                    <Form id="map-area-form" onSubmit={handleSubmit}>
+                        <Form.Group>
+                            <FinalInput name="name"
+                                        label={Translate.string('Name of the area')}
+                                        autoFocus
+                                        required />
+                        </Form.Group>
+                        <Form.Group>
+                            <FinalCheckbox name="default"
+                                           label={Translate.string('Default map area')}
+                                           toggle />
+                        </Form.Group>
+                        <Form.Group>
+                            <FinalInput name="bounds.southWest.lat"
+                                        type="number"
```

Comment:
`step="0.00000000000001"` - otherwise there are html5 validation errors as soon as you change the value manually (default step is 1)